website/docs/configuring_clients.md

```diff
@@ -341,9 +336,275 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
+  });
+}
+```
+
+### Receiving Push Notifications
+Receiving _Push Notifications_ is as simple as registering a callback handler:
+```javascript
+PushRegistration.onMessageReceived((notification => {
+  console.log('Received push notification: ', notification.message);
+}));
+```
+
+### Complete code
+Change the `src/index.js` file with the following content:
+```javascript
+import { PushRegistration } from '@aerogear/push';
+
+var app = {
+  // Application Constructor
+  initialize: function() {
+      document.addEventListener('deviceready', this.onDeviceReady.bind(this), false);
+  },
+
+  // deviceready Event Handler
+  //
+  // Bind any cordova events here. Common events are:
+  // 'pause', 'resume', etc.
+  onDeviceReady: function() {
+    this.receivedEvent('deviceready');
+  },
+
+  // Update DOM on a Received Event
+  receivedEvent: function(id) {
+    var parentElement = document.getElementById(id);
+    var listeningElement = parentElement.querySelector('.listening');
+    var receivedElement = parentElement.querySelector('.received');
+
+    listeningElement.setAttribute('style', 'display:none;');
+    receivedElement.setAttribute('style', 'display:block;');
+
+    this.registerToUPS();
+
+    PushRegistration.onMessageReceived((notification => {
+      console.log('Received push notification: ', notification.message);
+    }));
+  },
+
+  registerToUPS: () => {
+    console.log('Registering...');
+    new PushRegistration({
+      url: 'http://192.168.1.187:9999',  // change this to your UPS URL
+      android: { // your variant platform (android, ios or webpush)
+        senderID: '829475845435',  // Your senderID as you see it in your Firebase Console
+        variantID: '172bf953-f266-4e32-866b-662ff32d653c', // The id of the variant you created
+        variantSecret: '7680585e-c22e-4105-b0fc-fbcb150036d4' // the secret of the variant you created
+      }
+    })
+    .register()
+    .then(() => {
+      console.log('Registered!');
+    })
+    .catch(error => {
+      console.log('Failed: ', error.message, JSON.stringify(error))
+    });
+  }
+};
+
+app.initialize();
+``` 
+### Running the application
+The application can run into the emulator for Android, while it needs to run on a real device for iOS (iOS simulator 
+does not supports _push notifications_). 
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Android-->
+1. Be sure that the emulator is already up and running
+2. Issue `cordova emulate android`
+<!--iOS-->
+1. Connect an iOS device to the USB port of your mac
+1. Run _XCode_ with
+    ```bash
+    $ open platforms/ios/UpsHelloWorld.xcworkspace
+    ```
+2. Configure the _Signing and Capabilities_
+3. Click on the _run_ icon
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+#### Throubleshooting
+##### `cordova emulator android` takes forever to install the app into the emulator
+
+From time to time can happen that the `cordova emulator android` command is not able to install the app into the emulator.
```

Comment:
"This can happen when" instead of "From time to time can happen that"
Maybe "onto the emulator" instead of "into the emulator", but it can go either way.